### DISCUSSION TITLE
about bbox_iou

### DISCUSSION THREAD

**Discussion by John-chen521** [ultralytics/yolov5#2948] [Q&A]
I recently added SCA loss, but not work, because it is a bounding box loss function not based on IOU. Do I want to know if it has anything to do with the code

**Comment by glenn-jocher** [comment]
@John-chen521 hi! Thanks for the SCA loss suggestion. I'm not familiar with this, can you provide some sources and examples of how to implement? The current YOLOv5 box regression loss is based on CIoU and is computed in loss.py L133 here:
https://github.com/ultralytics/yolov5/blob/4890499344e21950d985e1a77e84a0a4161d1db0/utils/loss.py#L128-L134

**Reply by John-chen521** [reply]
Thank you very much.I'm still working on it

**Comment by Bara-Elba** [comment]
I want to evaluate my test set detections using IoU between the ground truth and the detected bounding boxes. to do so, I :
> 1. extracted the bboxes of the ground truth and the detection ones.
> 2. covert those coordination into a binary image.
> 3. calculate the IoU of the hole class

I am asking if there is a way to do it by calculating the IoU of the BBox with its corresponding BBox in the ground truth, because that would be more accurate.
Thanks in advance.

**Reply by glenn-jocher** [reply]
@Bara-Elba val.py evaluates metrics on any set in your dataset. To evaluate on test set:
```
python val.py --weights ... --data ... --task test
```
